actors/evm/src/interpreter/precompiles.rs

```diff
@@ -336,16 +349,49 @@ fn read_right_pad<'a>(input: impl Into<Cow<'a, [u8]>>, len: usize) -> Cow<'a, [u
 
 // --- Precompiles ---
 
-/// Read right padded BE encoded low u64 ID address from a u256 word
-/// returns encoded CID or an empty array if actor not found
-fn get_actor_code_cid<RT: Runtime>(
+/// Read right padded BE encoded low u64 ID address from a u256 word.
+/// Returns variant of [`BuiltinType`] encoded as a u256 word.
+fn get_actor_type<RT: Runtime>(
     system: &mut System<RT>,
     input: &[u8],
     _: PrecompileContext,
 ) -> PrecompileResult {
     let id_bytes: [u8; 32] = read_right_pad(input, 32).as_ref().try_into().unwrap();
     let id = Parameter::<u64>::try_from(&id_bytes)?.0;
-    Ok(system.rt.get_actor_code_cid(&id).unwrap_or_default().to_bytes())
+
+    if id >= 99 {
```

Comment:
Sorry _less_ than 99. I.e., all actors with IDs _less_ than 99 are "system".

Reply:
Er, actually, it turns out we have some accounts in this range (e.g., at 80).

Let's say < 32. That should be safe enough.

I.e., if the ID is < 32, return "1" (is system). Otherwise, look it up.